src/workerd/io/worker.c++

```diff
@@ -1652,6 +1652,16 @@ Worker::Worker(kj::Own<const Script> scriptParam,
             // evaluation is complete.
             TmpDirStoreScope tmpDirStoreScope;
 
+            // We allow eval and new Function() during startup, becaues startup time is entirely
+            // deterministic, so we can easily reproduce the input to eval() by just running the
+            // worker again. We do not allow eval() at runtime because we need to have a record of
+            // all code that executes in production for forensic purposes, and at runtime the input
+            // to eval() could have come from a remote source on which we don't have a record.
+            if (FeatureFlags::get(js).getAllowEvalDuringStartup()) {
+              js.setAllowEval(true);
+              KJ_DEFER(js.setAllowEval(false));
```

Comment:
Is that actually working? I would expect that to disable immediately because the scope ends immediately.

Reply:
So, this is why my feature flags check doesn't work...